Add blog post for Gateway API v0.5.0 beta release (#34968)
kubernetes/website: content/en/blog/_posts/2022-07-12-gateway-api-in-beta.md

(which represent the underlying network gateways / proxy servers) to enable
robust Kubernetes service networking through expressive, extensible and
role-oriented interfaces that are implemented by many vendors and have broad
industry support.

Originally conceived as a successor to the well known [Ingress][ing] API, the
benefits of Gateway API include (but are not limited to) explicit support for
many commonly used networking protocols (e.g. `HTTP`, `TLS`, `TCP`, `UDP`) as
well as tightly integrated support for Transport Layer Security (TLS). The
`Gateway` resource in particular enables implementations to manage the lifecycle
of network gateways as a Kubernetes API.

If you're an end-user interested in some of the benefits of Gateway API we
invite you to jump in and find an implementation that suits you. At the time of
this release there are over a dozen [implementations][impl] for popular API
gateways and service meshes and guides are available to start exploring quickly.

[gw]:https://gateway-api.sigs.k8s.io/api-types/gateway/
[ing]:https://kubernetes.io/docs/reference/kubernetes-api/service-resources/ingress-v1/

> [impl]:https://gateway-api.sigs.k8s.io/implementations

Reviewer: (nit)
```suggestion
[impl]:https://gateway-api.sigs.k8s.io/implementations/
```

Author: Updated, thanks!

> ### Getting Started

Reviewer: (nit)
```suggestion
### Getting started
```

Author: Updated, thanks!

The Gateway API project itself is a Kubernetes API like `Ingress` and you can
think of the implementations of it as similar to the implementations for
[IngressClass][ingc] or [StorageClass][strc] in that Kubernetes itself has no

> default implementation of it (though your cluster provider might).

Reviewer: Wouldn't Ingress be a better analogy here? There's no default Ingress implementation.

Reviewer: Anyway, it's better to get the thing published and worry about good docs later.

Author: Good call, reworked this section to focus on Ingress.

Take a look at our [API concepts documentation][concepts] and check out some of
our [Guides][guides] to start familiarizing yourself with the APIs and how they
work. When you're ready for a practical application open our [implementations
page][impl] and select an implementation that belongs to an existing technology
you may already be familiar with or the one your cluster provider uses as a
default (if applicable). Gateway API is a [Custom Resource Definition
(CRD)][crd] based API so you'll need to [install our CRDs][install-crds] onto a

> cluster to use the API.

Reviewer: In this paragraph, **our** has potentially switched from:
> Shane Utt (Kong), Rob Scott (Google), Nick Young (VMWare), Jeff Apple (Hashicorp)

_(the people who are pleased to be making this announcement)_

and has become, er
- _some SIG Network folk? I'm not sure_

How about rephrasing to “**the** API concepts documentation, **the** task-oriented Guides” &hellip; and “**the** CustomResourceDefinitions for this release”?

Author: Updated, thanks!

If you're specifically interested in helping to contribute to Gateway API, we
would love to have you! Please feel free to [open a new issue][issue] on the
repository, or join in our [discussions][disc]. Also check out our [community
page][community] which includes links to our Slack channel and community meetings.

[ingc]:https://kubernetes.io/docs/concepts/services-networking/ingress/#ingress-class
[strc]:https://kubernetes.io/docs/concepts/storage/storage-classes/
[crd]:https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/
[concepts]:https://gateway-api.sigs.k8s.io/concepts/api-overview/
[guides]:https://gateway-api.sigs.k8s.io/v1alpha2/guides/getting-started/
[impl]:https://gateway-api.sigs.k8s.io/implementations
[install-crds]:https://gateway-api.sigs.k8s.io/v1alpha2/guides/getting-started/#install-the-crds
[issue]:https://github.com/kubernetes-sigs/gateway-api/issues/new/choose

> [disc]:https://github.com/kubernetes-sigs/gateway-api/discussions/new

Reviewer: ```suggestion
[disc]:https://github.com/kubernetes-sigs/gateway-api/discussions/
```
It's better to let readers see existing topics than have them jump in to starting a fresh one without checking for dupes.

[community]:https://gateway-api.sigs.k8s.io/contributing/community/

## Release highlights

### Graduation to beta

The `v0.5.0` release is particularly historic because it marks the growth in

> maturity to a `v1beta1` release for some of our key APIs:

Reviewer: ```suggestion
maturity to a beta API version (`v1beta1`) for some of our key APIs:
```

Author: Updated, thanks!

- [GatewayClass](https://gateway-api.sigs.k8s.io/api-types/gatewayclass/)
- [Gateway](https://gateway-api.sigs.k8s.io/api-types/gateway/)
- [HTTPRoute](https://gateway-api.sigs.k8s.io/api-types/httproute/)

This achievement was marked by the completion of several graduation criteria:

- API has been [widely implemented][impl].
- Conformance tests provide basic coverage for all resources and have multiple implementations passing tests.
- Most of the API surface is actively being used.

> - Kubernetes SIG Network API reviewers have approved graduation to beta.

Reviewer: I still don't know whether or not Kubernetes' [API versioning guarantees](https://kubernetes.io/docs/reference/using-api/deprecation-policy/) apply to Gateway and friends. Maybe I don't get to find out (we can publish the release announcement without making that clear). Would be nice to know though.

Author: Good call, added a link to our versioning guidelines below this.

> To see what's in store for future releases check out the next steps section.

Reviewer: Should “next steps” be a hyperlink?

Author: Updated, thanks!
